### PR TITLE
samples: boards: nrf: nrfx_prs: extend regex to test default task

### PR DESCRIPTION
**Pull request by nordic-piks**
Verifies that at least SPIM is initialized properly.

### CODE REVIEW

**Review by jaz1-nordic — APPROVED** [review]

File: samples/boards/nrf/nrfx_prs/sample.yaml [hunk 1]
```diff
@@ -17,3 +17,8 @@ tests:
         - "nrfx PRS example on .*"
         - "-> press \".*\" to trigger a transfer"
         - "-> press \".*\" to switch the type of peripheral"
+        - "Switched to SPIM"
+        - "-- Background transfer on \".*\" --"
+        - "Tx: 4E 6F 72 64 69 63 20 53 65 6D 69 63 6F 6E 64 75 63 74 6F 72 00"
+        - "Rx:"
```
Comment (masz-nordic) [inline samples/boards/nrf/nrfx_prs/sample.yaml:23]: I would expect RX to be equal to TX?

**Review by masz-nordic — COMMENTED** [review]
I don't think this test is validating PRS properly.
First, the regex does not verify TX==RX.
Second, to fully test this functionality you would need to switch back and forth between SPIM and UARTE.
Checking if SPIM has initialized properly does not give much value and may lead to an invalid assumption that PRS is covered in verification.

File: samples/boards/nrf/nrfx_prs/sample.yaml (hunk 1 above)
Comment (nordic-piks) [reply samples/boards/nrf/nrfx_prs/sample.yaml:23]: Only if you have loopback connected. This seems to be optional for this sample. So I did not wanted to enforce that.

**Comment by nordic-piks** [comment]
> I don't think this test is validating PRS properly. First, the regex does not verify TX==RX. Second, to fully test this functionality you would need to switch back and forth between SPIM and UARTE. Checking if SPIM has initialized properly does not give much value and may lead to an invalid assumption that PRS is covered in verification.

I agree. However that require more effort to extend this test.
My idea was to extend regex to catch errors which already occurs, so only quick win.
Proper verification of PRS needs to be evaluated separately.

**Review by masz-nordic — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Review by nordic-segl — APPROVED** [review]

**Pull request merged**